### ISSUE TITLE
One-way sync

### ISSUE THREAD

**Issue by panhartstuff** [meganz/MEGAsync#227]
Is it possible to only sync one way; from PC to Mega? The reason is that  often make modifications to my folder but don't have access to MegaSync. So when I do get access to MegaSync, I want to make sure that old excess files from the older version on Mega don't get synced back to my PC which would cause a lot of major pain. Instead they'd just get deleted (on the Mega). 
Basically I want MegaSync to sync my folders, but in a way that does not make modifications to my local folder, without resyncing the whole thing again

**Comment by oSumAtrIX** [comment]
If you make changes locally they should be synced to the cloud too. Other way around is possible but a real solution to one way sync would be only possible if you permit file changes locally

**Comment by Trickster29** [comment]
I pay for mega pro iii and this would make my life much easier.

It prevents mega from redownloading files that I deleted or reverted to a previous version (Which sometimes it does do) and instead just treating the pc as the master copy. So if I revert a change. Mega doesn't revert the revert.

**Comment by zalxapuv** [comment]
I agree. This is something I would like too see implemented as well!

**Comment by Gerdal** [comment]
I agree too. It's big advantege of IceDrive!

**Comment by tio-trom** [comment]
I vote for this. I have a pro III account and millions of files that are constantly modified. I use Mega for backup. I only need my local files to be synced to Mega, not the other way around.

**Comment by idkman2021** [comment]
I vote for this!

**Comment by bcode2** [comment]
I vote for this too!. I just need  one way sync. Mega is always restoring files that I have previously deleted

**Comment by Wulgaren** [comment]
yes please, i'd like to save memory by deleting stuff from my computer but still make it upload whatever new stuff i have in my folder!

**Comment by thericle** [comment]
I hope this will get implemented, soon. As it is now, if you delete a file while MEGAsync is closed, it will come back when you open it again. This is rarely the desired behavior.

**Comment by idkman2021** [comment]
its frustrating because every once in a while MEGA sync gets confused about my external SSD and I have to reenable the sync folders by clicking checkmarks next to them, I use MEGA sync for keeping a backup of a lot of small files that often get deleted/reorganized and because of that behaviour and the amount of files oftentimes I have no clue if some old files have been redownloaded and messed up the organisation. I stopped trying to organize for the time being because checking everything would take too much time

**Comment by salvq** [comment]
I like mega but until one way sync is available I will not renew my subscription and I already switch to competitor…

**Comment by Wrizz** [comment]
It's been so long and it's STILL not fixed. 

I have been doing this in the meantime: have local folders somewhere on your drive and use something like "FreeFileSync" to sync them with MEGA folder and then check if anything is re-downloaded and overwritten, then delete the re-downloaded file and sync again, it won't re-download again until next time probably. File versioning or trash bin enabled doesn't help at all here.

**Comment by Myrkie** [comment]
I have to agree this feature is really really needed, I have mega syncing an entire 880GB archive for me and randomly it just "forgets" these are synced and ends up throwing my entire 880GB archive into the rubbish folder under a folder called "SyncDebris" which invalidates all generated links, I just want it to sync one way so i can have a backup and have another folder that I can use for uploading.

**Comment by dpe0** [comment]
Just noticed backup feature has been added and it seems to work as one way syns

**Comment by tio-trom** [comment]
> Just noticed backup feature has been added and it seems to work as one way syns

Can you give more info about it please? Where do I activate that or how does it work?

**Comment by Myrkie** [comment]
> > Just noticed backup feature has been added and it seems to work as one way syns
> 
> Can you give more info about it please? Where do I activate that or how does it work?

on the latest version of megasync (Atleast on windows) open open the settings from right clicking the tray and go to the backup tab
![image](https://user-images.githubusercontent.com/20288698/208959202-80616680-da35-4772-a7d2-dbc9a77440d4.png)

mega will also show all devices that run backups aswell as the backup state at https://mega.nz/#fm/devices

**Comment by tio-trom** [comment]
Ok interesting, thanks! Can't see that in Linux yet. Version 4.7.3.

**Comment by vtmateos** [comment]
Hi there,

You can check all the information in our blog post [here](https://blog.mega.io/introducing-mega-backup/). Backups functionality is available since v4.8.0.

Kind regards.

**Comment by agatakotecka** [comment]
> Hi there,
> 
> You can check all the information in our blog post [here](https://blog.mega.io/introducing-mega-backup/). Backups functionality is available since v4.8.0.
> 
> Kind regards.

Oh, okay that's good to hear!. But calling it 'backup' and making separate feature of it, is confusing and an overkill in my opinion.
You could have easily put an option under Sync tab, for user to decide which sync should be two-way sync and which one-way. That would make everyone's life easier.

**Comment by thericle** [comment]
One thing I like about MEGA's sync feature is that you can sync within the existing folder structure on the MEGA cloud. I'd like to be able to do that with the backup feature, too.

**Comment by Visual-Dawg** [comment]
And I would like to separate the backup from my PC ( being able to change the backup path ) as I want to backup photographs and they should not be treated as a PC backup, but as their own.

**Comment by VD171** [comment]
Any method for One-way sync, please?

**Comment by capturts** [comment]
@VD171 https://blog.mega.io/introducing-mega-backup/

**Comment by VD171** [comment]
> @VD171 https://blog.mega.io/introducing-mega-backup/

This does not allow files versioning.
File versioning is the reason I pay for mega pro.

**Comment by agatakotecka** [comment]
> > > Just noticed backup feature has been added and it seems to work as one way syns
> > 
> > 
> > Can you give more info about it please? Where do I activate that or how does it work?
> 
> on the latest version of megasync (Atleast on windows) open open the settings from right clicking the tray and go to the backup tab ![image](https://user-images.githubusercontent.com/20288698/208959202-80616680-da35-4772-a7d2-dbc9a77440d4.png)
> 
> mega will also show all devices that run backups aswell as the backup state at https://mega.nz/#fm/devices

I'm sorry but his is not a solution I'd expect from major leader in cloud computing, but looks rather like 'we have done something you can use, now f*ck off all you asking people'. 

People asking for one way sync have already their local files uploaded on MEGA. They don't need to create another 'Backup' cloud folder to sync them, they need to be able to tell MEGA app WHAT location should be synced. 

What we need to be implemented is to be able to sync any amount of local folder to user-chosen MEGA folders. 
This feature should also include filter function, to be able to sync SPECIFIC filetypes only, files meeting specific filesize criteria etc.....

What MEGA proposed above, is something rather I'd expect college student can do to pass exam, not from software developer !

**Comment by thericle** [comment]
@agatakotecka As much as I expected this feature, it was so unusable, that I had to go back to syncing.

I find two things very strange:
1. How did such a popular service not offer such a basic feature until recently (and now offer this half-baked solution)?
2. How difficult is it to keep everything the same and just give the option for the syncing to be one-way?

**Comment by agatakotecka** [comment]
> @agatakotecka As much as I expected this feature, it was so unusable, that I had to go back to syncing.
> 
> I find two things very strange:
> 
>     1. How did such a popular service not offer such a basic feature until recently (and now offer this half-baked solution)?
> 
>     2. How difficult is it to keep everything the same and just give the option for the syncing to be one-way?

@thericle Hello

1. That baffles me too. Too much time passed since one-way sync was requested, and in a meantime some 3rd party tools like Air Explorer offered it, make me think this is somehow deliberate by MEGA. Good developer would have done it already.

2. I don't think it's extremely difficult looking. It's about the will to do something, and there doesn't seem to be comprehensive approach to it by MEGA. Application development is always down to how good that person is.

**Comment by bcode2** [comment]
Because of  the lack  of interest in the implementation of one way sync  I am stopping the PRO plan and moving  all my data to  (gocryptfs + rsync+ onedrive). I have 6Tb  only 69Euros/year

**Comment by VD171** [comment]
> Because of the lack of interest in the implementation of one way sync I am stopping the PRO plan and moving all my data to (gocryptfs + rsync+ onedrive). I have 6Tb only 69Euros/year

For sure, I will not renew my mega account.
I need one-way sync for some folders I want to have an "exact backup".
I need symlinks for my whole ext4 partition.
Sad...

**Comment by qomhmd** [comment]
Any update on this? It is also OK to have the feature of keeping files on the cloud and request to download them when needed, like what Google Drive and Microsoft OneDrive provide (on Windows?).

**Comment by VictorGabriel7Dev** [comment]
> Any update on this? It is also OK to have the feature of keeping files on the cloud and request to download them when needed, like what Google Drive and Microsoft OneDrive provide (on Windows?).

You should give up.
Pay for a kvm, encrypt it and keep your backup files perfectly exact as you want.

**Comment by ciros88** [comment]
i would like to have real one-way sync (with support for file versioning), which is the main reason i pay for mega tbh, not for the current two-way sync
this has been asked in 2018 but still i don't see any progress here...
@vtmateos any plan for this?

**Comment by VD171** [comment]
> i would like to have real one-way sync (with support for file versioning), which is the main reason i pay for mega tbh, not for the current two-way sync this has been asked in 2018 but still i don't see any progress here... [@vtmateos](https://github.com/vtmateos) any plan for this?

You should give up.
Pay for a kvm, encrypt it and keep your backup files perfectly exact as you want.
